core/src/main/java/com/linecorp/armeria/server/healthcheck/DefaultHealthCheckUpdateHandler.java

```diff
@@ -0,0 +1,108 @@
+import com.fasterxml.jackson.databind.node.JsonNodeType;
+
+import com.linecorp.armeria.common.AggregatedHttpRequest;
+import com.linecorp.armeria.common.HttpRequest;
+import com.linecorp.armeria.common.HttpStatus;
+import com.linecorp.armeria.common.MediaType;
+import com.linecorp.armeria.server.HttpStatusException;
+import com.linecorp.armeria.server.ServiceRequestContext;
+
+final class DefaultHealthCheckUpdateHandler implements HealthCheckUpdateHandler {
+
+    static final DefaultHealthCheckUpdateHandler INSTANCE = new DefaultHealthCheckUpdateHandler();
+
+    private final ObjectMapper mapper = new ObjectMapper();
+
+    private DefaultHealthCheckUpdateHandler() {}
+
+    @Override
+    public CompletionStage<Boolean> handle(ServiceRequestContext ctx, HttpRequest req) throws Exception {
+        requireNonNull(req, "req");
+        switch (req.method()) {
+            case PUT:
+            case POST:
+                return req.aggregate().thenApply(this::handlePut);
+            case PATCH:
+                return req.aggregate().thenApply(this::handlePatch);
+            default:
+                throw HttpStatusException.of(HttpStatus.METHOD_NOT_ALLOWED);
+        }
+    }
+
+    private Boolean handlePut(AggregatedHttpRequest req) {
+        final JsonNode json = toJsonNode(req);
+        if (json.getNodeType() != JsonNodeType.OBJECT) {
+            throw HttpStatusException.of(HttpStatus.BAD_REQUEST);
+        }
+
+        final JsonNode healthy = json.get("healthy");
+        if (healthy == null) {
+            throw HttpStatusException.of(HttpStatus.BAD_REQUEST);
+        }
+        if (healthy.getNodeType() != JsonNodeType.BOOLEAN) {
+            throw HttpStatusException.of(HttpStatus.BAD_REQUEST);
+        }
+
+        return healthy.booleanValue();
+    }
+
+    private Boolean handlePatch(AggregatedHttpRequest req) {
+        final String json;
+        try {
+            json = mapper.writeValueAsString(toJsonNode(req));
+        } catch (JsonProcessingException e) {
+            throw new Error(e); // Never happens.
+        }
+
+        switch (json) {
+            case "[{\"op\":\"replace\",\"path\":\"/healthy\",\"value\":true}]":
+                return true;
+            case "[{\"op\":\"replace\",\"path\":\"/healthy\",\"value\":false}]":
+                return false;
+            default:
+                throw HttpStatusException.of(HttpStatus.BAD_REQUEST);
+        }
+    }
+
+    private JsonNode toJsonNode(AggregatedHttpRequest req) {
+        final MediaType contentType = req.contentType();
+        if (contentType != null && !contentType.is(MediaType.JSON)) {
+            throw HttpStatusException.of(HttpStatus.UNSUPPORTED_MEDIA_TYPE);
+        }
+
+        final Charset charset = contentType == null ? StandardCharsets.UTF_8
+                                                    : contentType.charset().orElse(StandardCharsets.UTF_8);
+        try {
+            return mapper.readTree(req.content(charset));
```

Comment:
Can consider calling `readTree(byte[])` if the charset is UTF_8 to avoid the copy for the vast majority case.

Reply:
Done